### PR TITLE
fix: invalid faculty number for attendance

### PR DESCRIPTION
**Pull request by Haider8**

### CODE REVIEW

**Comment by iamareebjamal** [comment]
You'll need to change the parsing below as well

**Comment by Haider8** [comment]
@iamareebjamal where?

File: app/src/main/java/amu/areeb/zhcet/utils/Utils.java [hunk 1]
```diff
@@ -23,16 +23,16 @@
...
-        else if (fcNo.length() != 8)
+        else if (fcNo.length() != 9)
             return false;
 
         String fYear = fcNo.substring(0, 2);
         String fBranch = fcNo.substring(2, 5);
         String fRNo = fcNo.substring(5);
 
-        return TextUtils.isDigitsOnly(fYear) && TextUtils.isDigitsOnly(fRNo) && fBranch.matches("^[ACEKLMP][EKR][B]$+") && Integer.parseInt(fYear) <= getSmallYear();
+        return TextUtils.isDigitsOnly(fYear) && TextUtils.isDigitsOnly(fRNo) && fBranch.matches("^[CEMP][ELKH][B]$+") && Integer.parseInt(fYear) <= getSmallYear();
```
Comment (Haider8) [inline app/src/main/java/amu/areeb/zhcet/utils/Utils.java:35]: If we have this return statement do we really need those if else conditions on length? Can't we add that condition here @iamareebjamal

Comment (iamareebjamal) [reply app/src/main/java/amu/areeb/zhcet/utils/Utils.java:35]: If these things are unnecessary, just remove the conditions on faculty number completely. Allow anything to be used as faculty number

Comment (Haider8) [reply app/src/main/java/amu/areeb/zhcet/utils/Utils.java:35]: These things are necessary. I will put the condition on faculty number length here which should be equal to 8

Comment (iamareebjamal) [reply app/src/main/java/amu/areeb/zhcet/utils/Utils.java:35]: But then I can't use the app as my faculty number is 14PEB049

Comment (Haider8) [reply app/src/main/java/amu/areeb/zhcet/utils/Utils.java:35]: So what should I do? Are we okay with current changes?

Comment (Haider8) [reply app/src/main/java/amu/areeb/zhcet/utils/Utils.java:35]: > But then I can't use the app as my faculty number is 14PEB049

You faculty number can't be found on the ctengg website.

Comment (iamareebjamal) [reply app/src/main/java/amu/areeb/zhcet/utils/Utils.java:35]: Yes, but it'll work in the app and API

Comment (iamareebjamal) [reply app/src/main/java/amu/areeb/zhcet/utils/Utils.java:35]: > So what should I do?

Remove all checks for faculty member or make it work for both kinds of faculty numbers

Comment (Haider8) [reply app/src/main/java/amu/areeb/zhcet/utils/Utils.java]: @iamareebjamal We can do one thing to make it work for all kinds of validations. Just removing the validation through branch regex and keeping all other things the same.

**Review by iamareebjamal — APPROVED** [review]

**Comment by Haider8** [comment]
@iamareebjamal Can we roll out a new version on the play store now?

**Comment by iamareebjamal** [comment]
Yes. Please update version name and code in build.gradle

**Pull request merged**